From a17b6a6045f75577342a3a6be9e64d9ed6a8de39 Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Wed, 9 Apr 2014 13:11:38 +0800
Subject: [PATCH 1555/1825] fix: sample at reset: could not set PCI-0/1 clock
 input/output mode

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit c807c4b685692a353addba7d3f241d4dfb279378

    pcie0/1 clock settings are enabled for DB-6660 and pcie0 is enabled for DB-6650.

Signed-off-by: Victor Gu <xigu@marvell.com>
Change-Id: Iab528c0a72f37c17c5033365d7cbd3468c5cd704
Reviewed-on: http://vgitil04.il.marvell.com:8080/7024
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h        |    6 ++++--
 .../avanta_lp_family/ctrlEnv/mvCtrlEnvSpec.h       |    2 ++
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h
index 2d6f5b5..8a67d66 100755
--- a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h
+++ b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h
@@ -79,6 +79,8 @@ typedef enum _mvSatRTypeID {
 	MV_SATR_CORE_CLK_SELECT,
 	MV_SATR_CPU1_ENABLE,
 	MV_SATR_SSCG_DISABLE,
+	MV_SATR_PEX0_CLOCK,
+	MV_SATR_PEX1_CLOCK,
 	// Jumpers  - change by removing Jumper. S@R will be changed by this option
 	MV_SATR_I2C0_SERIAL_ROM,
 	MV_SATR_EXTERNAL_CPU_RESET,
@@ -90,8 +92,6 @@ typedef enum _mvSatRTypeID {
 	MV_SATR_CPU0_ENDIANESS,
 	MV_SATR_CPU0_NMFI,
 	MV_SATR_CPU0_THUMB,
-	MV_SATR_PEX0_CLOCK,
-	MV_SATR_PEX1_CLOCK,
 	MV_SATR_REF_CLOCK_ENABLE,
 	MV_SATR_EFUSE_BYPASS,
 	MV_SATR_POR_BYPASS,
@@ -102,6 +102,8 @@ typedef enum _mvSatRTypeID {
 	MV_SATR_WRITE_CORE_CLK_SELECT,
 	MV_SATR_WRITE_CPU1_ENABLE,
 	MV_SATR_WRITE_SSCG_DISABLE,
+	MV_SATR_WRITE_PEX0_CLOCK,
+	MV_SATR_WRITE_PEX1_CLOCK,
 	MV_SATR_WRITE_MAX_OPTION
 } MV_SATR_TYPE_ID;
 
diff --git a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvSpec.h b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvSpec.h
index 661d778..35de0ab7 100644
--- a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvSpec.h
+++ b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvSpec.h
@@ -619,6 +619,8 @@ typedef enum _mvTarget {
 { MV_SATR_WRITE_CORE_CLK_SELECT, 0x00000001,	0,	1, {0, 1, 0, 1} },\
 { MV_SATR_WRITE_CPU1_ENABLE,	 0x00000002,	1,	1, {0, 1, 0, 1} },\
 { MV_SATR_WRITE_SSCG_DISABLE,	 0x00000004,	2,	1, {0, 1, 0, 1} },\
+{ MV_SATR_WRITE_PEX0_CLOCK,	 0x00000008,	3,	1, {0, 1, 0, 1} },\
+{ MV_SATR_WRITE_PEX1_CLOCK,	 0x00000010,	4,	1, {0, 0, 0, 1} },\
 };
 
 #endif /* MV_ASMLANGUAGE */
-- 
1.7.5.4

